src/installer.ts

```diff
@@ -1,54 +1,103 @@
+import os = require('os');
 import * as assert from 'assert';
 import * as core from '@actions/core';
 import * as hc from '@actions/http-client';
 import * as io from '@actions/io';
 import * as tc from '@actions/tool-cache';
-import * as os from 'os';
 import * as path from 'path';
 import * as semver from 'semver';
-
-let osPlat: string = os.platform();
-let osArch: string = translateArchToDistUrl(os.arch());
+import {Url} from 'url';
+import fs = require('fs');
 
 //
 // Node versions interface
 // see https://nodejs.org/dist/index.json
 //
-interface INodeVersion {
+export interface INodeVersion {
   version: string;
   files: string[];
 }
 
-export async function getNode(versionSpec: string) {
+interface INodeVersionInfo {
```

Comment:
These interfaces normalize the resolution from manifest and fallback to node dist so common code downstream can be agnostic.